File: second/secondasync/src/main/java/joe/async/mapper/SecondMapper.java
package joe.async.mapper;

import joe.reactive.dto.ThirdDto;
import joe.model.SecondDto;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface SecondMapper {
    default ThirdDto map(SecondDto secondDto) {
        return ThirdDto.builder()
                .field111(secondDto.getField11())
                .field222(secondDto.getField22())
                .field333(secondDto.getField33())
                .field444(secondDto.getField44())
                .needCallback(secondDto.isNeedCallback())
                .build();
    }
}

File: second/secondredis/src/main/java/redis/mapper/SecondMapper.kt
package redis.mapper

import joe.model.SecondDto
import joe.reactive.dto.ThirdDto
import org.springframework.stereotype.Component

@Component
class SecondMapper {
    fun map(secondDto: SecondDto): ThirdDto {
        return ThirdDto.builder()
            .field111(secondDto.field11)
            .field222(secondDto.field22)
            .field333(secondDto.field33)
            .field444(secondDto.field44)
            .build()
    }
}